### PR TITLE
#19 extend PgUserITCase

### PR DESCRIPTION
**Pull request by olenagerasimova**
This is for #19.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#47](https://github.com/g4s8/ghman/issues/47) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#47](https://github.com/g4s8/ghman/issues/47) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CJT4R32Q3/c54ba5d0-bc7b-41e1-a3e3-7054672a1c5a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/com/g4s8/ghman/data/PgUserITCase.java [hunk 1]
```diff
@@ -16,32 +16,123 @@
...
+        ).affirm();
+    }
+
+    /**
+     * Configures {@link Chat} mock for tests.
+     * @param tid Telegtam id
+     * @param umane User name
+     * @return Mocked {@link Chat} instance
+     */
+    private static Chat getChat(final long tid, final String umane) {
```
Comment (victornoel) [inline src/test/java/com/g4s8/ghman/data/PgUserITCase.java]: @olenagerasimova please at least rename this method to `chatMock`

File: src/test/java/com/g4s8/ghman/data/PgUserITCase.java [hunk 2]
```diff
@@ -16,32 +16,123 @@
...
+        final String uname = "test4";
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(tid, uname));
+        new Assertion<>(
+            "Should return json with telegram name and id",
+            user.telegram(),
+            new AllOf<>(
+                new IterableOf<>(
+                    new JsonHas("firstName", new JsonValueIs(uname)),
+                    new JsonHas("uid", new JsonValueIs(tid))
```
Comment (victornoel) [inline src/test/java/com/g4s8/ghman/data/PgUserITCase.java]: @olenagerasimova isn't there 2 other field in this json datastructure?

File: src/test/java/com/g4s8/ghman/data/PgUserITCase.java [hunk 3]
```diff
@@ -16,32 +16,123 @@
...
+    @Test
+    void returnsGhUser() throws IOException {
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(408L, "test2"));
+        final String token = "some_token";
+        user.authorize(token);
+        new Assertion<>(
+            new FormattedText(
+                "Should return github API instance with token %s", token
+            ).toString(),
```
Comment (victornoel) [inline src/test/java/com/g4s8/ghman/data/PgUserITCase.java]: @olenagerasimova please use `asString()` here (and maybe it's not worh the complexification to build a proper string like this, by reading the test things already clear.

File: src/test/java/com/g4s8/ghman/data/PgUserITCase.java (hunk 2 above)
Comment (olenagerasimova) [reply src/test/java/com/g4s8/ghman/data/PgUserITCase.java]: @victornoel no, my mocked chat has only first name and user id

**Comment by olenagerasimova** [comment]
@victornoel done, take a look please

**Comment by 0crat** [comment]
@victornoel[/z](https://www.0crat.com/u/victornoel) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/test/java/com/g4s8/ghman/data/PgUserITCase.java (hunk 2 above)
Comment (victornoel) [reply src/test/java/com/g4s8/ghman/data/PgUserITCase.java]: @olenagerasimova ah, I understand, yep it makes sense

**Comment by victornoel** [comment]
@g4s8 looks good to me

**Comment by olenagerasimova** [comment]
@g4s8 ping
@victornoel have already approved PR and I've merged master

**Comment by victornoel** [comment]
@olenagerasimova I think it is better to use rebase instead of merge for PRs

**Comment by victornoel** [comment]
@olenagerasimova ping, please rebase so that the commit history is readable

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (9 days), architects (@g4s8) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#47](https://github.com/g4s8/ghman/issues/47) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ammaratef45** [comment]
@victornoel You should find at least three problems in the PR.

Please confirm you will try to find three or more problems next time.

**Comment by victornoel** [comment]
@ammaratef45 yes

**Comment by ammaratef45** [comment]
@victornoel thanks

**Comment by ammaratef45** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Pull request merged**